tests/test_fhir_client_patient_delete.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any, Optional
+
+import requests
+import requests_mock
+from requests import Response
+
+from helix_fhir_client_sdk.fhir_client import FhirClient
+
+
+def test_fhir_client_patient_delete() -> None:
+    with requests_mock.Mocker() as mock:
+        # Arrange
+        url = "http://foo"
+
+        def delete_matcher(request: Any) -> Optional[Response]:
+            if request.path == "/patient/12345":
+                resp: Response = requests.Response()
+                resp.status_code = 204
```

Comment:
the possible correct status codes for response for DELTE could be multiple;
200 OK or 204 No Content, when resource deleted successfully,
202 Accepted, when resource was marked for deletion for getting it queued.
FYI